contracts/NFTVesting/LinearVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.5.0) (token/ERC20/utils/TokenTimelock.sol)
+// SpartanLabs Contracts (NFTVesting)
+
+pragma solidity ^0.8.0;
+import "./IERC721.sol";
+
+/**
+ * @dev A single NFT holder contract that will allow a beneficiary to extract the
+ * NFT after a given vesting start time.
+ * After the vesting start time, the discount will start to accumulate for the locker linearly.
+ *
+ * Developers would have to perform the following actions for the locking of NFT:
+ * Deploy -> Approve -> Transfer
+ *
+ * Note that in order for discount in ETH to be valid, ETH must first be sent to this contract upon token locking.
+ */
+contract LinearVestingNFTTimeLock {
+    // ERC721 basic token smart contract
+    IERC721 private immutable _nft;
+
+    // ERC721 basic token ID of contract being held
+    uint256 private immutable _tokenId;
+
+    // beneficiary of token after they are released
+    address private immutable _beneficiary;
+
+    // timestamp when token release is enabled and when discount starts to vest. Also known as cliff period.
+    uint256 private immutable _vestingStartTime;
+
+    // Max discount allowed for a token in percentage
+    uint256 private immutable _maxDiscountPercentage;
```

Comment:
Same, if we already have max duration and amount of ETH into this contract, that should suffice to calculate max discount, having an additional variable seems to be mistake inducing - is there a reason we still need this? Perhaps I'm missing something in my thought process here

Reply:
thanks for this comment, removed max discount to simplify things